apps/dashboard/src/contract-ui/tabs/settings/components/metadata.tsx

```diff
@@ -63,27 +60,26 @@ function extractDomain(url: string) {
   }
 }
 
-export const SettingsMetadata = <
-  TContract extends ValidContractInstance | undefined,
->({
+export const SettingsMetadata = ({
   contract,
   detectedState,
 }: {
-  contract: TContract;
+  contract: ThirdwebContract;
   detectedState: ExtensionDetectedState;
 }) => {
   const trackEvent = useTrack();
-  const metadata = useContractMetadata(contract);
+  const metadata = useReadContract(getContractMetadata, { contract });
   const sendTransaction = useSendAndConfirmTransaction();
   const invalidateContract = useInvalidatev4Contract();
 
   const transformedQueryData = useMemo(() => {
     return {
       ...metadata.data,
       name: metadata.data?.name || "",
+      image: metadata.data?.image || "",
       dashboard_social_urls: Object.entries(
         metadata.data?.social_urls || { twitter: "", discord: "" },
-      ).map(([key, value]) => ({ key, value })),
+      ).map(([key, value]) => ({ key, value: String(value) })),
```

Comment:
why is this required, what else other than string would it be?

Reply:
This was to make typescript happy as the type here is unknown. I can try to find a better approach

Reply:
what's the type of `metadata.data.social_urls` ?

Reply:
`any`. Because here's the type of `metadata.data` from v5:

```ts
(property) data: {
    [key: string]: any;
    name: string;
    symbol: string;
} | undefined
```

Reply:
Do we update the type in SDK?

Reply:
no, you'll probably have to add a typeguard here, so a function that takes in `metadata` and validates that the keys have certain shapes etc (you could use zod for this) and then returns it back out